crates/forge/src/shared_cache.rs

```diff
@@ -0,0 +1,53 @@
+pub const CACHE_DIR: &str = ".snfoundry_cache";
+
+use anyhow::Result;
+use camino::Utf8PathBuf;
+use scarb_metadata::MetadataCommand;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader, Write};
+
+use crate::test_case_summary::TestCaseSummary;
+
+pub fn read_tests_failed_file() -> Result<Vec<String>> {
+    let tests_failed_path = cache_dir()?.join(".prev_tests_failed");
```

Comment:
Preferebly we can have `".prev_tests_failed"` as a global `const &str`, then we can reuse it below too